gulpfile.js/build.js

```diff
@@ -392,10 +463,404 @@ function buildPages(done) {
+  );
+  env.addFilter('importBlog', importBlog, true);
+
+  env.addFilter('importYouTubeChannel', importYouTubeChannel, true);
+  env.addFilter('survey', survey, true);
+
+  return env;
+}
+
+const getUpdatedURL = (u, requestedFormat, forcedFormat) => {
+  return u.replace(
+    /(.*documentation\/[^/]+)[\/.]([^?]+)(?:\?(?:[^=]*)=(.*))?/,
+    (match, section, page) => {
+      // covering the case where /documentation/tools.html, which matches differently than all other paths
+      if (page === 'html') {
+        page = 'index.html';
+      }
+
+      const hasFormat = forcedFormat || requestedFormat;
+      const fmt = hasFormat ? `${hasFormat}/` : '';
+      return `${section}/${fmt}${page}`;
+    }
+  );
+};
+
+function optimizeFiles(cb) {
+  const logger = require('@lib/utils/log')('AMP Optimizer');
+  return gulp
+    .src(`${project.paths.PAGES_DEST}/**/*.html`)
+    .pipe(
+      through.obj((file, encoding, callback) => {
+        const unoptimizedFile = file.contents.toString();
+
+        optimize({query: ''}, unoptimizedFile, {}).then((optimizedFile) => {
+          file.contents = Buffer.from(optimizedFile);
+          callback(null, file);
+        });
+      })
+    )
+    .pipe(
+      gulp.dest((f) => {
+        logger.log(
+          `staticified ${path.relative(project.absolute('.'), f.path)}`
+        );
+        return f.base;
+      })
+    )
+    .on('end', cb);
+}
+
+function newPost(text, img, id) {
+  return {
+    id: id,
+    text: text,
+    img: '/static/samples/img/' + img,
+    timestamp: Number(new Date()),
+  };
+}
+
+async function renderExamples(done) {
+  const logger = require('@lib/utils/log')('Static File Generator');
+  const env = nunjucksEnv();
+  const blogItems = [
+    newPost('A green landscape with trees.', 'landscape_green_1280x853.jpg', 1),
+    newPost(
+      'Mountains reflecting on a lake.',
+      'landscape_mountains_1280x657.jpg',
+      2
+    ),
+    newPost(
+      'A road leading to a lake with mountains on the back.',
+      'landscape_lake_1280x857.jpg',
+      3
+    ),
+    newPost(
+      'Forested hills with a grey sky in the background.',
+      'landscape_trees_1280x960.jpg',
+      4
+    ),
+    newPost(
+      'Scattered houses in a mountain village.',
+      'landscape_village_1280x853.jpg',
+      5
+    ),
+    newPost('A deep canyon.', 'landscape_canyon_1280x1700.jpg', 6),
+    newPost(
+      'A desert with mountains in the background.',
+      'landscape_desert_1280x853.jpg',
+      7
+    ),
+    newPost('Colorful houses on a street.', 'landscape_houses_1280x803.jpg', 8),
+    newPost('Blue sea surrounding a cave.', 'landscape_sea_1280x848.jpg', 9),
+    newPost(
+      'A ship sailing the sea at sunset.',
+      'landscape_ship_1280x853.jpg',
+      10
+    ),
+  ];
+
+  const configObj = {
+    time: new Date().toLocaleTimeString(),
+    timestamp: Number(new Date()),
+    // send a random list of blog items to make it also work on the cache
```

Comment:
does this still make sense?

Reply:
I can double check if it is nolonger needed, but I believe it needs to be there

The rendering of the pages happens in two steps - the initial render (that would normally be handled by grow at serve time). That step outputs all of the files, which include a handful of files that have our own templating inside of it that is rendered with nunjucks currently. The `blogItems` bit is there for that initial render. I could easily be wrong, though.

Reply:
What I mean is that the random part does not make sense if we're rendering at build time